src/node_crypto.cc

```diff
@@ -4656,10 +4657,33 @@ void ECDH::GenerateKeys(const FunctionCallbackInfo<Value>& args) {
 
   if (!EC_KEY_generate_key(ecdh->key_))
     return env->ThrowError("Failed to generate EC_KEY");
-
-  ecdh->generated_ = true;
 }
 
+void ECDH::GeneratePublicKey(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  ECDH* ecdh = Unwrap<ECDH>(args.Holder());
+
+  const BIGNUM* priv = EC_KEY_get0_private_key(ecdh->key_);
+  if (priv == nullptr)
+    return env->ThrowError("Failed to get ECDH private key");
+
+  EC_POINT* pub = EC_POINT_new(ecdh->group_);
+  if (pub == nullptr)
+    return env->ThrowError("Failed to allocate EC_POINT for a public key");
+
+  if (EC_POINT_mul(ecdh->group_, pub, priv, nullptr, nullptr, nullptr) <= 0) {
+    EC_POINT_free(pub);
+    return env->ThrowError("Failed to generate ECDH public key");
+  }
+
+  if (!EC_KEY_set_public_key(ecdh->key_, pub)) {
+    EC_POINT_free(pub);
+    return env->ThrowError("Failed to convert EC_POINT to a public key");
+  }
+
+  EC_POINT_free(pub);
```

Comment:
I'm confused, isn't this function supposed to return a formatted public key, or is the added documentation wrong?